server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -219,6 +223,34 @@ public BalancerSegmentHolder pickSegmentToMove(final List<ServerHolder> serverHo
     return sampler.getRandomBalancerSegmentHolder(serverHolders);
   }
 
+  @Override
+  public Iterator<ServerHolder> pickServersToDrop(DataSegment toDrop, NavigableSet<ServerHolder> serverHolders)
+  {
+    List<ListenableFuture<Pair<Double, ServerHolder>>> futures = Lists.newArrayList();
+
+    for (final ServerHolder server : serverHolders) {
+      futures.add(
+          exec.submit(
+              () -> Pair.of(computeCost(toDrop, server, true), server)
+          )
+      );
+    }
+
+    final ListenableFuture<List<Pair<Double, ServerHolder>>> resultsFuture = Futures.allAsList(futures);
+
+    try {
+      List<Pair<Double, ServerHolder>> results = resultsFuture.get();
+      return results.stream()
```

Comment:
This code is slightly obtuse (what's the double? why reverse it?) and would benefit from a comment.